tests/e2e/system_test/test_full_cluster_health.py

```diff
@@ -0,0 +1,286 @@
+        """
+
+        mgr_pod_obj = pod.get_mgr_pods()
+        mgr_node_obj = pod.get_pod_node(mgr_pod_obj[0])
+
+        self.nodes.restart_nodes([mgr_node_obj])
+
+        wait_for_nodes_status()
+
+        # Check for Ceph pods
+        pod_obj = ocp.OCP(kind=constants.POD, namespace=defaults.ROOK_CLUSTER_NAMESPACE)
+        assert pod_obj.wait_for_resource(
+            condition="Running", selector="app=rook-ceph-mgr", timeout=600
+        )
+        assert pod_obj.wait_for_resource(
+            condition="Running",
+            selector="app=rook-ceph-mon",
+            resource_count=3,
+            timeout=600,
+        )
+        assert pod_obj.wait_for_resource(
+            condition="Running",
+            selector="app=rook-ceph-osd",
+            resource_count=3,
+            timeout=600,
+        )
+
+    def restart_ocs_operator_node(self):
+        """
+        Restart node that runs OCS operator pod
+        """
+
+        pod_obj = pod.get_ocs_operator_pod()
+        node_obj = pod.get_pod_node(pod_obj)
+
+        self.nodes.restart_nodes([node_obj])
+
+        wait_for_nodes_status()
+
+        pod.wait_for_pods_to_be_running(
+            namespace=constants.OPENSHIFT_STORAGE_NAMESPACE, pod_names=[pod_obj.name]
+        )
+
+    def is_cluster_healthy(self):
+        """
+        Wrapper function for cluster health check
+
+        Returns:
+            bool: True if ALL checks passed, False otherwise
+        """
+        return self.ceph_not_health_error() and pod.wait_for_pods_to_be_running()
+
+    @tier3
+    def test_full_cluster_health(self, nodes):
+        """
+        Verify that the cluster health is ok when the storage is ~75% full
+
+        """
+        self.nodes = nodes
+
+        self.full_log_path = get_full_test_logs_path(cname=self)
+        logger.info(f"Logs file path name is : {self.full_log_path}")
+
+        logger.info("Create resource file for fio workload")
+        self.crd_data = templating.load_yaml(constants.FIO_CR_YAML)
+
+        self.set_storageclass(interface=constants.CEPHBLOCKPOOL)
+
+        self.calculate_crd_data()
+
+        self.run()
+
+        logger.info("Checking health before disruptive operations")
+        assert self.is_cluster_healthy(), "Cluster is not healthy"
+
+        osd_node_reboot()
+        logger.info("Checking health after OSD node reboot")
+        assert self.is_cluster_healthy(), "Cluster is not healthy"
+
+        self.mgr_pod_node_restart()
+        logger.info("Checking health after worker node shutdown")
+        assert self.is_cluster_healthy(), "Cluster is not healthy"
+
+        self.restart_ocs_operator_node()
+        logger.info("Checking health after OCS operator node restart")
+        assert self.is_cluster_healthy(), "Cluster is not healthy"
+
+        self.delete_pods()
+        logger.info("Checking health after Rook, OSD, MGR & MON pods deletion")
+        assert self.is_cluster_healthy(), "Cluster is not healthy"
+
```

Comment:
lets also do some sanity validation of creating and deleting resources.
You can use Sanity class for doing these.

Reply:
Done